### PR TITLE
Update tsconfig and make nav interfaces exposable for consumption

### PR DESCRIPTION
**Pull request by sandrahoang686**
**Related Ticket:** https://github.com/NASA-IMPACT/veda-ui/issues/1186
**Related PR:** https://github.com/developmentseed/next-veda-ui/pull/10

**Note**
To expose interfaces from veda-ui - they should be defined in `.ts` files vs `.d.ts` files. This is because parcel isn't building the typescript definition files into `lib/index.d.ts` but does so when the file is `.ts`. 

**When building the library with `.d.ts` extension - there are NO type hints from NextInstance side**
Screenshot 2024-11-04 at 9 23 19 AM

**When building the library with `.ts` extension - there are type hints form NextInstance side**
Screenshot 2024-11-04 at 9 20 02 AM

### Description of Changes
* Update typescript definition file of nav types to typescript file
* Add declaration true to tsconfig

### Notes & Questions About Changes
Found this thread about typescript definition files and parcel build here https://github.com/parcel-bundler/parcel/issues/1240

### Validation / Testing
_{Update with info on what can be manually validated in the Deploy Preview link for example "Validate style updates to selection modal do NOT affect cards"}_

### CODE REVIEW

**Review by dzole0311 — APPROVED** [review]

**Comment by dzole0311** [comment]
Thanks for fixing, I'm getting the correct type annotations now ✅

**Review by hanbyul-here — APPROVED** [review]
I checked the built asset (`lib/index.d.ts`) has Nav related types. thanks for working on it 🙇 Maybe easy lint warnings can be fixed (ex. linebreaks) can be fixed since we are on it? but not a blocker !

File: tsconfig.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
     "moduleResolution": "node",
     "esModuleInterop": true,
     "strictNullChecks": true,
+    "declaration": true,
```
Comment (hanbyul-here) [inline tsconfig.json:8]: for my education, can you tell me what this line does?

Comment (sandrahoang686) [reply tsconfig.json:8]: here is more about it https://www.typescriptlang.org/tsconfig/declaration.html

**Pull request merged**